### PR TITLE
Tpetra,MueLu: GPU fixes

### LINKED ISSUE #8955: MueLu:  printout for coarse type: none
## Bug Report
@trilinos/muelu 

### Description
When running a multilevel method with `coarse type: none` there is a warning message printed every iteration that you have "No coarse grid solver". Is this necessary? It is kind of annoying when you are running without a coarse gird on purpose. 

Related, why does `coarse type: none` not match something less elegant like a Jacobi relaxation with a damping parameter of 0.0 on the coarse grid?

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/tpetra @trilinos/muelu 

## Motivation
GPU related changes:
- Tpetra: Make use of TPL provided spmv in `Tpetra::residual` a runtime option
- MueLu: Don't set `Klu` options for non-Klu solvers (e.g.CuSOLVER)
- MueLu: Fix bad PL with Tpetra matrixmatrix options
- MueLu: Only warn in setup about levels without smoother/coarse solver. Fixes #8955

### CODE REVIEW

File: packages/tpetra/core/src/Tpetra_Details_residual.hpp [hunk 1]
```diff
@@ -338,16 +338,18 @@ void localResidual(const CrsMatrix<SC,LO,GO,NO> &  A,
...
-#else
+
+  const bool useNativeSpMVInResidual = ::Tpetra::Details::Behavior::useNativeSpMVInResidual ();
+  if (useNativeSpMVInResidual) {
+    SC one = Teuchos::ScalarTraits<SC>::one();
+    SC negone = -one;
+    SC zero = Teuchos::ScalarTraits<SC>::zero();
+    // This is currently a "reference implementation" waiting until Kokkos Kernels provides
+    // a residual kernel.
+    A.localApply(X_colmap,R,Teuchos::NO_TRANS, one, zero);
```
Comment (brian-kelley) [inline packages/tpetra/core/src/Tpetra_Details_residual.hpp:349]: I know this codepath predates your changes, but ``A.localApply`` is still going to use the default spmv (a TPL if available). If there's a need to use native spmv in Tpetra, it should probably be an optional parameter to localApply. When the KokkosKernels 4.3 promotion is merged, this path could pass in ``KokkosSparse::SPMV_NATIVE``). Then localApply would create a temporary handle if the algorithm is anything other than ``SPMV_DEFAULT``.

BTW, is a TPL implementation causing problems or showing bad performance? Or is this just about replacing a fixed macro with a Behavior option?

Comment (brian-kelley) [reply packages/tpetra/core/src/Tpetra_Details_residual.hpp:349]: I misunderstood the intent of this behavior, it really means "use a non-fused residual" so the codepath is correct for that.

File: packages/muelu/src/MueCentral/MueLu_Hierarchy_def.hpp [hunk 2]
```diff
@@ -834,7 +847,6 @@ ConvergenceStatus Hierarchy<Scalar, LocalOrdinal, GlobalOrdinal, Node>::Iterate(
     emptyFineSolve = false;
   }
   if (emptyFineSolve == true) {
-    GetOStream(Warnings1) << "No fine grid smoother" << std::endl;
```
Comment (GrahamBenHarper) [inline packages/muelu/src/MueCentral/MueLu_Hierarchy_def.hpp:837]: rip

**Review by GrahamBenHarper — APPROVED** [review]
looks good to me!

**Pull request merged**